File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:

    int lower_bound(vector<int>& nums, int target){
        int st = 0 ,en = nums.size()-1;
        int mid;
        while(st<=en){
            mid = (st+en)/2;

            if(nums[mid]<target){
                st = mid+1;
            }else{
                en = mid-1;
            }
        }
        return st;
    }

    vector<int> searchRange(vector<int>& nums, int target) {
        
        vector<int> res(2,-1);
        int lowerInd = lower_bound(nums,target);
        int higherInd = lower_bound(nums,target+1)-1;

        if(lowerInd < nums.size()&&target == nums[lowerInd]){
            res[0] = lowerInd;
            res[1] = higherInd;
        }
        return res;
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {

    int findnextgreaterelement(int[] nums,int target){
        int st = 0,en=nums.length;
        int mid;
        while(st<en){
            mid = (st+en)/2;
            if(nums[mid]<=target){
                st = mid+1;
            }else{
                en = mid;
            }
        }
        return st;
    }

    public int[] searchRange(int[] nums, int target) {
        int lastindex = findnextgreaterelement(nums,target);
        int firstindex = findnextgreaterelement(nums,target-1);
        
        int[] ans = new int[2];
        if(lastindex==0 || nums[lastindex-1]!=target){
            ans[0]=-1;ans[1]=-1;
        }else{
            ans[0]=firstindex;
            ans[1]=lastindex-1;
        }
        return ans;
    }
}